scripts/tools/silabs/FactoryDataProvider.py

```diff
@@ -0,0 +1,212 @@
+import subprocess
+import sys
+import argparse
+import struct
+
+# global variables
+Discriminator_ui = 0
+Passcode_ui = 0
+IterationCount_ui = 0
+Salt_str = ""
+Verifier_str = ""
+mcuFamily = ""
+jtagSerial = ""
+isDeviceConnected = True
+
+# CONSTANTS
+TEMP_FILE = 'tmp_nvm3.s37'
+OUT_FILE = 'matter_factorydata.s37'  # Final output file containing the nvm3 data to flash to the device
+# nvm3 keys to set
+PASSCODE_NVM3_KEY = "0x87205:"
+DISCRIMINATOR_NVM3_KEY = "0x87207:"
+ITERATIONCOUNT_NVM3_KEY = "0x87208:"
+SALT_NVM3_KEY = "0x87209:"
+VERIFIER_NVM3_KEY = "0x8720A:"
+
+
+def generate_spake2p_verifier(args):
+    """ Generate Spake2+ verifier using the external spake2p tool
+
+    Args:
+        The whole set of args passed to the script. The required one are:
+        gen_spake2p_path: path to spake2p executable
+        spake2_iteration: Iteration counter for Spake2+ verifier generation
+        passcode: Pairing passcode using in Spake2+
+        spake2_salt: Salt used to generate Spake2+ verifier
+
+    Returns:
+        The generated verifier string
+    """
+
+    cmd = [
+        args.gen_spake2p_path, 'gen-verifier',
+        '--iteration-count', str(args.spake2_iteration),
+        '--salt', args.spake2_salt,
+        '--pin-code', str(args.passcode),
+        '--out', '-',
+    ]
+    output = subprocess.check_output(cmd)
+    output = output.decode('utf-8').splitlines()
+    generation_results = dict(zip(output[0].split(','), output[1].split(',')))
+    return generation_results["Verifier"]
+
+
+def processArguments(args):
+    """ Do some checks on the received arguments.
+        Generate the Spake2+ verifier if needed and assign the values
+        to the global variables
+
+    Args:
+        The whole set of args passed to the script.
+    """
+    global Discriminator_ui
+    global Passcode_ui
+    global IterationCount_ui
+    global Salt_str
+    global Verifier_str
+    global mcuFamily
+    global jtagSerial
```

Comment:
No need to write "global" in front of each of these. They are in scope from being in same module.